flash/core/data/io/input_base.py

```diff
@@ -0,0 +1,173 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import functools
+import sys
+from typing import Any, Iterable, MutableMapping, Sequence, Union
+
+from torch.utils.data import Dataset, IterableDataset
+
+from flash.core.data.data_pipeline import DataPipeline
+from flash.core.data.properties import Properties
+from flash.core.utilities.stages import RunningStage
+
+if sys.version_info < (3, 7):
+    from typing import GenericMeta
+else:
+    GenericMeta = type
+
+
+def _has_len(data: Union[Sequence, Iterable]) -> bool:
+    try:
+        len(data)
+        return True
+    except (TypeError, NotImplementedError):
+        return False
+
+
+def _validate_input(input: "InputBase"):
+    if input.data is not None:
```

Comment:
None is considered a valid value since we use it to determine whether the `Input` is truthy or falsey which is then used by the datamodule to know whether to provide a dataset / dataloader for that stage.